pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,237 @@
+	if err != nil {
+		t.Fatalf("can't create new encoded replace certificate: %v", err)
+	}
+	if err = createFileWithContent(wo.TLSCertFile, encodedReplaceCert.cert); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+	if err = createFileWithContent(wo.TLSKeyFile, encodedReplaceCert.privateKey); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+
+	err = wait.PollImmediate(1*time.Second, 3*time.Second, func() (bool, error) {
+		resp, err := client.Get(addr)
+		return err == nil && reflect.DeepEqual(resp.TLS.PeerCertificates[0].SerialNumber, replaceCertSerialNumber), nil
+	})
+	if err != nil {
+		t.Errorf("certificate wasn't updated: %v", err)
+	}
+}
+
+type CertificateCreator struct {
+	ca           *x509.Certificate
+	caPrivateKey *rsa.PrivateKey
+}
+
+type encodedCertificate struct {
+	cert       []byte
+	privateKey []byte
+}
+
+func newCertificateCreator() (*CertificateCreator, error) {
+	now := time.Now()
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(0),
+		NotBefore:    now,
+		NotAfter:     now.Add(1 * time.Hour),
+
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+		IsCA:                  true,
+
+		DNSNames: []string{"localhost"},
+	}
+
+	caPrivateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CertificateCreator{
+		ca:           ca,
+		caPrivateKey: caPrivateKey,
+	}, nil
+}
+
+func (cc *CertificateCreator) getEncodedCA() ([]byte, error) {
+	bytes, err := x509.CreateCertificate(rand.Reader, cc.ca, cc.ca, cc.caPrivateKey.Public(), cc.caPrivateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return pemEncode(bytes, "CERTIFICATE")
+}
+
+func (cc *CertificateCreator) newEncodedCertificate(serialNumber *big.Int) (*encodedCertificate, error) {
+	now := time.Now()
+	cert := &x509.Certificate{
+		SerialNumber: serialNumber,
+		NotBefore:    now,
+		NotAfter:     now.Add(1 * time.Hour),
+
+		KeyUsage:    x509.KeyUsageDigitalSignature,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+
+		DNSNames: []string{"localhost"},
+	}
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, err
+	}
+
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, cc.ca, privateKey.Public(), cc.caPrivateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	certEncoded, err := pemEncode(certBytes, "CERTIFICATE")
+	if err != nil {
+		return nil, err
+	}
+
+	privateKeyEncoded, err := pemEncode(x509.MarshalPKCS1PrivateKey(privateKey), "RSA PRIVATE KEY")
+	if err != nil {
+		return nil, err
+	}
+
+	return &encodedCertificate{
+		cert:       certEncoded,
+		privateKey: privateKeyEncoded,
+	}, nil
+}
+
+func pemEncode(cert []byte, certType string) ([]byte, error) {
+	buff := new(bytes.Buffer)
```

Comment:
tiny nit: for consistency - let's use `buff := &bytes.Buffer{}` or `var buff bytes.Buffer`